massa-execution/src/interface_impl.rs

```diff
@@ -27,17 +36,117 @@ impl InterfaceClone for InterfaceImpl {
 
 impl Interface for InterfaceImpl {
     fn get_module(&self, address: &String) -> Result<Vec<u8>> {
-        let bytecode = {
-            let address = Address::from_str(address)?;
-            let context = self
-                .context
-                .lock()
-                .expect("Failed to acquire lock on context.");
-            context.ledger_step.get_module(&address)
-        };
+        let address = Address::from_str(address)?;
+        let bytecode = { context_guard!(self).ledger_step.get_module(&address) };
         match bytecode {
-            Some(bytecode) => Ok(bytecode),
+            Some(bytecode) => {
+                context_guard!(self).call_stack.push_back(address);
```

Comment:
why are we pushing the address on the call stack here ? we just want to read the module from an address

Reply:
This is a naming issue. When we get a module with the interface it's necessary to execute it

Reply:
uuh

Reply:
get_module should be get_module_for_execution

Reply:
hmmm... let's discuss that :)

Reply:
Can we close the thread?

Reply:
ok